### PR TITLE
No longer return incorrect keys in `attributes` result of `getModelDataWithAttributes` test util.

### LINKED ISSUE #19757: `getModelDataWithAttributes` handles empty string attributes incorrectly
## 📝 Provide detailed reproduction steps (if any)

1. Create a model element with two attributes:
    * One attribute with an empty string value (e.g., `htmlContent: ''`).
    * Another attribute with an object value (e.g., `htmlInputAttributes: { classes: ... }`).
2. Run `getModelDataWithAttributes()` on this model.
3. Check the returned `data` string and the `attributes` object.

### ✔️ Expected result

The function should consistently reference attributes by their ID. If the empty string is stored in the `attributes` object at index `1`, the data string should refer to it as `(1)`, and subsequent attributes should follow the sequence (e.g., `(2)`).

```js
{
    data: '<paragraph><htmlInput htmlContent="(1)" htmlInputAttributes="(2)"></htmlInput></paragraph>',
    attributes: {
        1: '',
        2: {
            classes: [ 'foo', 'bar' ]
        }
    }
}
```

### ❌ Actual result

The empty string is inlined in the `data` string (as `htmlContent=""`), but it still consumes an index in the `attributes` object (`1: ''`).

This causes a mismatch for subsequent attributes. The second attribute in the string uses `(1)`, which points to the empty string in the object, instead of pointing to the correct object data (which is stored at index `2`).

```js
{
    data: '<paragraph><htmlInput htmlContent="" htmlInputAttributes="(1)"></htmlInput></paragraph>',
    attributes: {
        1: '', // Assigned ID 1, but not referenced in data string
        2: {
            classes: [ 'foo', 'bar' ] // Assigned ID 2, but data string refers to (1)
        }
    }
}
```

### PR DESCRIPTION
**Pull request by Mati365**
### 🚀 Summary

No longer return incorrect keys in `attributes` result of `getModelDataWithAttributes` test util.

---

### 📌 Related issues

* Closes #19757 

---

### 🧾 Checklists

Use the following checklists to ensure important areas were not overlooked.
This does not apply to feature-branch merges.
If an item is **not relevant** to this type of change, simply leave it unchecked.

#### Author checklist

- [x] Is the changelog entry intentionally omitted?
- [x] Is the change backward-compatible?
- [ ] Have you considered the impact on different editor setups and core interactions? _(e.g., classic/inline/multi-root/many editors, typing, selection, paste, tables, lists, images, collaboration, pagination)_
- [ ] Has the change been manually verified in the relevant setups?
- [ ] Does this change affect any of the above?
- [ ] Is performance impacted?
- [ ] Is accessibility affected?
- [ ] Have tests been added that fail without this change (against regression)?
- [ ] Have the API documentation, guides, feature digest, and related feature sections been updated where needed?
- [ ] Have metadata files (ckeditor5-metadata.json) been updated if needed?
- [ ] Are there any changes the team should be informed about (e.g. architectural, difficult to revert in future versions or having impact on other features)?
- [ ] Were these changes documented (in Logbook)?

#### Reviewer checklist

- [ ] PR description explains the changes and the chosen approach (especially when  performance, API, or UX is affected).
- [ ] The changelog entry is clear, user‑ or integrator-facing, and it describes any breaking changes.
- [ ] All new external dependencies have been approved and mentioned in LICENSE.md (if any).
- [ ] All human-readable, translateable strings in this PR been introduced using `t()` (if any).
- [ ] I manually verified the change (e.g., in manual tests or documentation).
- [ ] The target branch is correct.

### CODE REVIEW

File: packages/ckeditor5-html-support/tests/_utils/utils.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ export function getModelDataWithAttributes( model, options = {} ) {
 	let attributes = [];
 	for ( const item of range.getItems() ) {
 		for ( const [ key, value ] of sortAttributes( item.getAttributes() ) ) {
-			if ( key.startsWith( 'html' ) && !excludeAttributes.includes( key ) ) {
+			if ( key.startsWith( 'html' ) && !excludeAttributes.includes( key ) && typeof value === 'object' ) {
```
Comment (gorzelinski) [inline packages/ckeditor5-html-support/tests/_utils/utils.js]: I don't know if the edge case applies, but `typeof null === 'object'`. Maybe:

```suggestion
			if ( key.startsWith( 'html' ) && !excludeAttributes.includes( key ) && value && typeof value === 'object' ) {
```
